### ISSUE TITLE
`prefer-string-slice` - false positive for Math.max

### ISSUE THREAD

**Issue by Maxim-Mazurok** [sindresorhus/eslint-plugin-unicorn#2021]
I've had this code:
```js
function createNonce() {
  var s = "",
    length = 32;
  do {
    s += Math.random().toString(36).substr(2);
  } while (s.length < length);
  s = s.substr(0, length);
  return s;
}
```
the tool with recommended config changed it to this:
```js
function createNonce() {
  var s = "",
    length = 32;
  do {
    s += Math.random().toString(36).slice(2);
  } while (s.length < length);
  s = s.slice(0, Math.max(0, length));
  return s;
}
```

As you can see, it added `s.slice(0, Math.max(0, length)` even though `length` is a "constant" number. I believe `isLengthProperty` function might've been tricked by the variable name `length`, not property `"string".length`.

`prefer-string-slice`

**Comment by fisker** [comment]
Let's not make this rule complex, if you feel this output is odd, we can change it to suggestion instead of auto-fix.

**Comment by Maxim-Mazurok** [comment]
Yeah, I found it quite odd and manually changed to  `s.slice(0, length)`. Up to you what to do about it, I've just started using this plugin today.

Unrelated: so far there were a lot of questionable changes, but some did help to catch pretty nasty stuff. Overall I'm quite pleased with the defaults, but my mistake was trying to apply all recommended rules at once on a large code base, making code review a pain. Next time I will be applying rules from the recommended list on by one.

**Comment by fregante** [comment]
Works as intended. `Math.max()` specifically exists to make the auto-fix as safe as possible.

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/891842d539d0d002119e9c476a050a12d3936a75/rules/prefer-string-slice.js#L56

This could be a _enhancement_: if the number is a constant, don't use `Math.max()`.

> Let's not make this rule complex

👍 closing then.

**Comment by Maxim-Mazurok** [comment]
Fair enough, especially since you never really know if something is constant in JS... Thanks for considering!
